client/js/components/main/assessment.jsx

```diff
@@ -49,79 +96,18 @@ export class Assessment extends React.Component{
   }
 
   render(){
-    window.onbeforeunload = this.popup;
```

Comment:
This has all been refactored to methods defined on the class. A call to ```this.content()``` should contain all of this logic.